homeassistant/components/sensor/opentherm_gw.py

```diff
@@ -0,0 +1,209 @@
+        gw_vars.DATA_DHW_PUMP_STARTS: [None, None, "Hot Water Pump Starts"],
+        gw_vars.DATA_DHW_BURNER_STARTS: [
+            None, None, "Hot Water Burner Starts"],
+        gw_vars.DATA_TOTAL_BURNER_HOURS: [
+            None, UNIT_HOUR, "Total Burner Hours"],
+        gw_vars.DATA_CH_PUMP_HOURS: [
+            None, UNIT_HOUR, "Central Heating Pump Hours"],
+        gw_vars.DATA_DHW_PUMP_HOURS: [None, UNIT_HOUR, "Hot Water Pump Hours"],
+        gw_vars.DATA_DHW_BURNER_HOURS: [
+            None, UNIT_HOUR, "Hot Water Burner Hours"],
+        gw_vars.DATA_MASTER_OT_VERSION: [
+            None, None, "Thermostat OpenTherm Version"],
+        gw_vars.DATA_SLAVE_OT_VERSION: [
+            None, None, "Boiler OpenTherm Version"],
+        gw_vars.DATA_MASTER_PRODUCT_TYPE: [
+            None, None, "Thermostat Product Type"],
+        gw_vars.DATA_MASTER_PRODUCT_VERSION: [
+            None, None, "Thermostat Product Version"],
+        gw_vars.DATA_SLAVE_PRODUCT_TYPE: [None, None, "Boiler Product Type"],
+        gw_vars.DATA_SLAVE_PRODUCT_VERSION: [
+            None, None, "Boiler Product Version"],
+        gw_vars.OTGW_MODE: [None, None, "Gateway/Monitor Mode"],
+        gw_vars.OTGW_DHW_OVRD: [None, None, "Gateway Hot Water Override Mode"],
+        gw_vars.OTGW_ABOUT: [None, None, "Gateway Firmware Version"],
+        gw_vars.OTGW_BUILD: [None, None, "Gateway Firmware Build"],
+        gw_vars.OTGW_CLOCKMHZ: [None, None, "Gateway Clock Speed"],
+        gw_vars.OTGW_LED_A: [None, None, "Gateway LED A Mode"],
+        gw_vars.OTGW_LED_B: [None, None, "Gateway LED B Mode"],
+        gw_vars.OTGW_LED_C: [None, None, "Gateway LED C Mode"],
+        gw_vars.OTGW_LED_D: [None, None, "Gateway LED D Mode"],
+        gw_vars.OTGW_LED_E: [None, None, "Gateway LED E Mode"],
+        gw_vars.OTGW_LED_F: [None, None, "Gateway LED F Mode"],
+        gw_vars.OTGW_GPIO_A: [None, None, "Gateway GPIO A Mode"],
+        gw_vars.OTGW_GPIO_B: [None, None, "Gateway GPIO B Mode"],
+        gw_vars.OTGW_SB_TEMP: [
+            DEVICE_CLASS_TEMPERATURE, TEMP_CELSIUS,
+            "Gateway Setback Temperature"],
+        gw_vars.OTGW_SETP_OVRD_MODE: [
+            None, None, "Gateway Room Setpoint Override Mode"],
+        gw_vars.OTGW_SMART_PWR: [None, None, "Gateway Smart Power Mode"],
+        gw_vars.OTGW_THRM_DETECT: [None, None, "Gateway Thermostat Detection"],
+        gw_vars.OTGW_VREF: [None, None, "Gateway Reference Voltage Setting"],
+    }
+    sensors = []
+    for var in discovery_info:
+        device_class = sensor_info[var][0]
+        unit = sensor_info[var][1]
+        friendly_name = sensor_info[var][2]
+        sensors.append(
+            OpenThermSensor(hass, var, device_class, unit, friendly_name))
+    async_add_entities(sensors)
+
+
+class OpenThermSensor(Entity):
+    """Representation of an OpenTherm Gateway sensor."""
+
+    def __init__(self, hass, var, device_class, unit, friendly_name):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
Pass in the entity_id instead.